politeiad/backendv2/tstorebe/store/mysql/mysql.go

```diff
@@ -0,0 +1,304 @@
+// Copyright (c) 2020-2021 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package mysql
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"path/filepath"
+	"sync"
+	"time"
+
+	"github.com/decred/politeia/politeiad/backendv2/tstorebe/store"
+	"github.com/decred/politeia/util"
+
+	_ "github.com/go-sql-driver/mysql"
+)
+
+const (
+	// Database options
+	connTimeout     = 1 * time.Minute
+	connMaxLifetime = 1 * time.Minute
+	maxOpenConns    = 0 // 0 is unlimited
+	maxIdleConns    = 100
+
+	// Database table names
+	tableNameKeyValue = "kv"
+	tableNameNonce    = "nonce"
+)
+
+// tableKeyValue defines the key-value table.
+const tableKeyValue = `
+  k VARCHAR(38) NOT NULL PRIMARY KEY,
```

Comment:
Don't size varchar.

Reply:
MySQL does not allow for an unsized varchar.  It has been updated to `VARCHAR(255)`.

https://dev.mysql.com/doc/refman/5.7/en/char.html

https://github.com/decred/politeia/pull/1180/files#diff-9cbdfdfae5a29aa39168b4484642d0acb8fdd880dcccdb358538928ac60cdb70R33